content/getting-started/go.textile

```diff
@@ -0,0 +1,266 @@
+
+go get -u github.com/ably/ably-go/ably
+```
+
+<aside data-type='note'>
+<p>The code examples in this guide include a demo API key. If you wish to interact with the Ably CLI and view outputs within your Ably account, ensure that you replace them with your own API key.</p>
+</aside>
+
+h2(#step-1). Step 1: Connect to Ably
+
+Clients establish a connection with Ably when they instantiate an SDK instance. This enables them to send and receive messages in realtime across channels.
+
+* Create a @main.go@ file in your project and add the following function to instantiate the SDK and establish a connection to Ably. At the minimum you need to provide an authentication mechanism. Use an API key for simplicity, but you should use token authentication in a production app. A @clientId@ ensures the client is identified, which is required to use certain features, such as presence:
+
+```[go]
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/ably/ably-go/ably"
+)
+
+func main() {
+	client, err := ably.NewRealtime(
+		ably.WithKey("{{API_KEY}}"),
+		ably.WithClientID("my-first-client"),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	// Wait for the connection to be connected
+	connStateChan := make(chan ably.ConnectionStateChange, 1)
+	client.Connection.On(ably.ConnectionEventConnected, func(change ably.ConnectionStateChange) {
+		connStateChan <- change
+	})
+	select {
+	case <-connStateChan:
+		fmt.Println("Made my first connection!")
+	case <-context.Background().Done():
+		log.Fatal("Context cancelled before connection established")
+	}
+
+	// Keep the program running
+	select {}
+}
+```
+
+You can monitor the lifecycle of clients' connections by registering a listener that will emit an event every time the connection state changes. For now, run the function with @go run main.go@ to log a message to the console to know that the connection attempt was successful. You'll see the message printed to your console, and you can also inspect the connection event in the dev console of your app.
+
+h2(#step-2). Step 2: Subscribe to a channel and publish a message
+
+Messages contain the data that a client is communicating, such as a short 'hello' from a colleague, or a financial update being broadcast to subscribers from a server. Ably uses channels to separate messages into different topics, so that clients only ever receive messages on the channels they are subscribed to.
+
+* Update your @main()@ function to create a channel instance and register a listener to subscribe to the channel. Then run it with @go run main.go@:
+
+```[go]
+func main() {
+	// ...existing code...
+
+	channel := client.Channels.Get("my-first-channel")
+
+	// Subscribe to messages
+	unsubscribe, err := channel.Subscribe(context.Background(), "", func(msg *ably.Message) {
+		fmt.Printf("Received message: %s\n", msg.Data)
+	})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer unsubscribe()
+
+	// Keep the program running
+	select {}
+}
```

Comment:
It is somewhat obvious, but we have `// ...existing code...` above but have repeated the closing lines.